hkmc2/shared/src/main/scala/hkmc2/bbml/bbML.scala

```diff
@@ -337,11 +345,14 @@ class BBTyper(using elState: Elaborator.State, tl: TL):
         ascribe(lam, sigTy)
         ()
       case N =>
-        val funTy = freshVar
-        pctx += sym -> funTy // for recursive types
+        given Ctx = ctx.nextLevel
+        val funTyV = freshVar
+        pctx += sym -> funTyV // for recursive functions
         val (res, _) = typeCheck(lam)
+        val funTy = monoOrErr(res, lam)
```

Comment:
Why would a general type raise an error here? If this case is not supposed to happen, assert so.

Reply:
Is this resolved? If so, please mark so.